orekit/tests/validate_frames.py

```diff
@@ -0,0 +1,225 @@
+
+import orekit
+from orekit.pyhelpers import setup_orekit_curdir
+
+# Setup orekit virtual machine and associated data
+VM = orekit.initVM()
+setup_orekit_curdir("orekit-data.zip")
+
+# All interesting 3D directions
+R_SET, V_SET = [product([-1, 0, 1], repeat=3) for _ in range(2)]
+
+# Retrieve celestial bodies from orekit
+Sun_orekit = CelestialBodyFactory.getSun()
+Mercury_orekit = CelestialBodyFactory.getMercury()
+Venus_orekit = CelestialBodyFactory.getVenus()
+Earth_orekit = CelestialBodyFactory.getEarth()
+Mars_orekit = CelestialBodyFactory.getMars()
+Jupiter_orekit = CelestialBodyFactory.getJupiter()
+Saturn_orekit = CelestialBodyFactory.getSaturn()
+Uranus_orekit = CelestialBodyFactory.getUranus()
+Neptune_orekit = CelestialBodyFactory.getNeptune()
+
+# Name of the bodies
+BODIES_NAMES = [
+    "Sun",
+    "Mercury",
+    "Venus",
+    "Earth",
+    "Mars",
+    "Jupiter",
+    "Saturn",
+    "Uranus",
+    "Neptune",
+]
+
+# orekit: bodies, inertial and fixed frames
+OREKIT_BODIES = [
+    Sun_orekit,
+    Mercury_orekit,
+    Venus_orekit,
+    Earth_orekit,
+    Mars_orekit,
+    Jupiter_orekit,
+    Saturn_orekit,
+    Uranus_orekit,
+    Neptune_orekit,
+]
+OREKIT_FIXED_FRAMES = [body.getBodyOrientedFrame() for body in OREKIT_BODIES]
+
+# poliastro: bodies, inertial and fixed frames
+POLIASTRO_BODIES = [
+    Sun_poliastro,
+    Mercury_poliastro,
+    Venus_poliastro,
+    Earth_poliastro,
+    Mars_poliastro,
+    Jupiter_poliastro,
+    Saturn_poliastro,
+    Uranus_poliastro,
+    Neptune_poliastro,
+]
+POLIASTRO_ICRS_FRAMES = [
+    HCRS_poliastro,
+    MercuryICRS_poliastro,
+    VenusICRS_poliastro,
+    GCRS_poliastro,
+    MarsICRS_poliastro,
+    JupiterICRS_poliastro,
+    SaturnICRS_poliastro,
+    UranusICRS_poliastro,
+    NeptuneICRS_poliastro,
+]
+POLIASTRO_FIXED_FRAMES = [
+    SunFixed_poliastro,
+    MercuryFixed_poliastro,
+    VenusFixed_poliastro,
+    ITRS_poliastro,
+    MarsFixed_poliastro,
+    JupiterFixed_poliastro,
+    SaturnFixed_poliastro,
+    UranusFixed_poliastro,
+    NeptuneFixed_poliastro,
+]
+
+
+# Collect both API data in two dictionaries
+OREKIT_BODIES_AND_FRAMES = dict(
+    zip(BODIES_NAMES, zip(OREKIT_BODIES, OREKIT_FIXED_FRAMES))
+)
+POLIASTRO_BODIES_AND_FRAMES = dict(
+    zip(
+        BODIES_NAMES,
+        zip(POLIASTRO_BODIES, POLIASTRO_ICRS_FRAMES, POLIASTRO_FIXED_FRAMES),
+    )
+)
+
+
+# Macros for J2000 and ICRF frame from Orekit API
+J2000_OREKIT = AbsoluteDate.J2000_EPOCH
+ICRF_FRAME_OREKIT = FramesFactory.getICRF()
+
+
+@pytest.mark.parametrize("r_vec", R_SET)
+@pytest.mark.parametrize("v_vec", V_SET)
+@pytest.mark.parametrize("body_name", BODIES_NAMES)
+def validate_from_body_intertial_to_body_fixed(body_name, r_vec, v_vec):
+
+    # Conversion frames for these bodies are expected to fail since Orekit 10.3
+    # makes use of IAU WGCCRE 2009 report while poliastro uses IAU WGCCRE 2015
+    # report
+    if body_name in ["Mercury", "Mars", "Neptune"]:
```

Comment:
Let's parametrize this and use `pytest.param` instead, see https://github.com/poliastro/poliastro/search?q=pytest.param